### ISSUE TITLE
TUS uploads do not honor WebDAV destination paths and get stored into the user's base path

### ISSUE THREAD

**Issue by SamuAlfageme** [owncloud/client#8053]
### Expected behavior
When creating a folder sync connection to a remote subdirectory, that subdirectory should be used as base path for the TUS uploads. 

### Actual behavior
Neither the `'Upload-Metadata": filename'` header **nor the WebDAV endpoint** (`<server>/remote.php/dav/files/<user>/`) on the `POST` request made by `PropagateUploadFileTUS::makeCreationWithUploadJob()` contain the full destination path:

https://github.com/owncloud/client/blob/2763c78abdc33c7d0415e111360a0b506064fcf2/src/libsync/propagateuploadtus.cpp#L79-L87

This results on all the uploads to **custom WebDAV paths** to end up being uploaded to the user's home instead.

### Steps to reproduce
1. Create the following file structure:

```console
$ tree
.
└── subdir
    └── test.txt

1 directory, 1 file
```

2. Create a destination `<subfolder>` on the server:

```console
$ curl -s -k -X MKCOL https://<user>:<pass>@<server>/remote.php/webdav/<subfolder>
```

3. Synchronize the contents of the working directory with the remote `<subfolder>` by running the client:

```console 
$ QT_LOGGING_RULES='*=false;sync.httplogger=true' \
  /usr/bin/cernboxcmd --trust $(pwd) \
  ownclouds://<user>:<pass>@<server>/remote.php/webdav/<subfolder>
```

4. List **recursively** (notice the depth) the contents of `<subfolder>`. The `subdir` folder was created, but empty (as `MKCOL` is independent from TUS):

```console
$ curl -s -k -H "Depth: infinity" -X PROPFIND https://<user>:<pass>@<server>/remote.php/webdav/<subfolder> | xmllint --format -
<?xml version="1.0" encoding="utf-8"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns">
  <d:response>
    <d:href>/remote.php/webdav/<subfolder>/</d:href>
    <d:propstat>
      <d:prop>
        <oc:id>MTI4NGQyMzgtYWE5Mi00MmNlLWJkYzQtMGIwMDAwMDA5MTU4OjE0NDY4MDI4MA==</oc:id>
        <oc:fileid>MTI4NGQyMzgtYWE5Mi00MmNlLWJkYzQtMGIwMDAwMDA5MTU4OjE0NDY4MDI4MA==</oc:fileid>
        <d:getetag>"89fa558:1599519376.363"</d:getetag>
        <oc:permissions>WCKDNVR</oc:permissions>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <d:getcontenttype>httpd/unix-directory</d:getcontenttype>
        <oc:size>0</oc:size>
        <d:getlastmodified>Mon, 07 Sep 2020 22:56:16 +0000</d:getlastmodified>
        <oc:favorite>0</oc:favorite>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
  <d:response>
    <d:href>/remote.php/webdav/<subfolder>/subdir/</d:href>
    <d:propstat>
      <d:prop>
        <oc:id>MTI4NGQyMzgtYWE5Mi00MmNlLWJkYzQtMGIwMDAwMDA5MTU4OjE0NDY4MDI4MQ==</oc:id>
        <oc:fileid>MTI4NGQyMzgtYWE5Mi00MmNlLWJkYzQtMGIwMDAwMDA5MTU4OjE0NDY4MDI4MQ==</oc:fileid>
        <d:getetag>"89fa559:1599519376.363"</d:getetag>
        <oc:permissions>WCKDNVR</oc:permissions>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <d:getcontenttype>httpd/unix-directory</d:getcontenttype>
        <oc:size>0</oc:size>
        <d:getlastmodified>Mon, 07 Sep 2020 22:56:16 +0000</d:getlastmodified>
        <oc:favorite>0</oc:favorite>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
</d:multistatus>
```

5. List the `subdir` folder, also created on the base path. Which contains the original files:

```console
$ curl -s -k -X PROPFIND https://<user>:<pass>@<server>/remote.php/webdav/subdir | xmllint --format -
<?xml version="1.0" encoding="utf-8"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns">
  <d:response>
    <d:href>/remote.php/webdav/subdir/</d:href>
    <d:propstat>
      <d:prop>
        <oc:id>MTI4NGQyMzgtYWE5Mi00MmNlLWJkYzQtMGIwMDAwMDA5MTU4OjE0NDY4MDI4Mg==</oc:id>
        <oc:fileid>MTI4NGQyMzgtYWE5Mi00MmNlLWJkYzQtMGIwMDAwMDA5MTU4OjE0NDY4MDI4Mg==</oc:fileid>
        <d:getetag>"89fa55a:1599519490.165"</d:getetag>
        <oc:permissions>WCKDNVR</oc:permissions>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <d:getcontenttype>httpd/unix-directory</d:getcontenttype>
        <oc:size>11</oc:size>
        <d:getlastmodified>Mon, 07 Sep 2020 22:58:10 +0000</d:getlastmodified>
        <oc:favorite>0</oc:favorite>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
  <d:response>
    <d:href>/remote.php/webdav/subdir/test.txt</d:href>
    <d:propstat>
      <d:prop>
        <oc:id>MTI4NGQyMzgtYWE5Mi00MmNlLWJkYzQtMGIwMDAwMDA5MTU4OjY4MjcyNTAxMzQxNzA5OTI2NA==</oc:id>
        <oc:fileid>MTI4NGQyMzgtYWE5Mi00MmNlLWJkYzQtMGIwMDAwMDA5MTU4OjY4MjcyNTAxMzQxNzA5OTI2NA==</oc:fileid>
        <d:getetag>"682725013417099264:00000001"</d:getetag>
        <oc:permissions>WCKDNVR</oc:permissions>
        <d:getcontentlength>0</d:getcontentlength>
        <d:getcontenttype>application/octet-stream</d:getcontenttype>
        <d:getlastmodified>Mon, 07 Sep 2020 22:58:09 +0000</d:getlastmodified>
        <oc:favorite>0</oc:favorite>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
</d:multistatus>
```

### Server configuration

Tested against our fork of [`eos-ocis:latest`](https://hub.docker.com/r/owncloud/eos-ocis); built in pointing to @ishank011's fork of [`ocis-reva`](https://github.com/ishank011/ocis-reva/). But this rather a client issue.

<details>
<summary><code>GET '/ocs/v1.php/cloud/capabilities?format=json'</code></summary>

```json
{
    "ocs": {
        "data": {
            "capabilities": {
                "checksums": {
                    "preferredUploadType": "SHA256",
                    "supportedTypes": [
                        "SHA256"
                    ]
                },
                "core": {
                    "pollinterval": 60,
                    "status": {
                        "edition": "community",
                        "installed": true,
                        "maintenance": false,
                        "needsDbUpgrade": false,
                        "productname": "reva",
                        "version": "10.0.11.5",
                        "versionstring": "10.0.11"
                    },
                    "support-url-signing": false,
                    "webdav-root": "remote.php/webdav"
                },
                "dav": {
                    "chunking": "",
                    "chunkingParallelUploadDisabled": false,
                    "reports": [
                        "search-files"
                    ],
                    "trashbin": "1.0"
                },
                "files": {
                    "bigfilechunking": false,
                    "blacklisted_files": [],
                    "privateLinks": false,
                    "tus_support": {
                        "extension": "creation,creation-with-upload",
                        "http_method_override": "",
                        "max_chunk_size": 0,
                        "resumable": "1.0.0",
                        "version": "1.0.0"
                    },
                    "undelete": true,
                    "versioning": true
                },
                "files_sharing": {
                    "api_enabled": true,
                    "auto_accept_share": true,
                    "default_permissions": 22,
                    "federation": {
                        "incoming": true,
                        "outgoing": true
                    },
                    "group_sharing": true,
                    "public": {
                        "enabled": true,
                        "expire_date": {
                            "enabled": true
                        },
                        "multiple": true,
                        "password": {
                            "enforced": true,
                            "enforced_for": {
                                "read_only": true,
                                "read_write": true,
                                "upload_only": true
                            }
                        },
                        "send_mail": true,
                        "social_share": true,
                        "supports_upload_only": true,
                        "upload": true
                    },
                    "resharing": true,
                    "search_min_length": 3,
                    "share_with_group_members_only": true,
                    "share_with_membership_groups_only": true,
                    "user": {
                        "send_mail": true
                    },
                    "user_enumeration": {
                        "enabled": true,
                        "group_members_only": true
                    }
                },
                "notifications": {
                    "ocs-endpoints": [
                        "disable"
                    ]
                }
            },
            "version": {
                "edition": "reva",
                "major": 10,
                "micro": 11,
                "minor": 0,
                "string": "10.0.11"
            }
        },
        "meta": {
            "message": "OK",
            "status": "ok",
            "statuscode": 100
        }
    }
}
```
</details>

### Client configuration

```console
cernbox version 2.7.0v270beta1 (build 1913)
Git revision 9b1d812ea160edef096d982cdb4140f206576bdb
Using Qt 5.12.9, built against Qt 5.12.9
Using 'OpenSSL 1.0.2k-fips  26 Jan 2017'
```

### Logs

```logs
09-07 09:54:07:329 [ info sync.httplogger ]:	"7969c301-21f2-485f-95a2-31e5905d994a: 
  Request: POST https://<user>:<pass>@<server>/remote.php/dav/files/<user>/ 
  Header: 
    {
      "X-OC-Mtime":1599465190,
      "Content-Type":"application/offset+octet-stream",
      "Content-Length":8,
      "Upload-Offset":0,
      "Tus-Resumable":1.0.0,
      "Upload-Metadata":filename dGV4dC50eHQ=,checksum U0hBMjU2IDEzNTUwMzUwYTg2ODFjODRjODYxYWFjMmU1YjQ0MDE2MWMyYjMzYTNlNGYzMDJhYzY4MGNhNWI2ODZkZTQ4ZGU=,
      "Upload-Length":8,
      "Authorization":[
          "redacted"
      ],
      "User-Agent":Mozilla/5.0 (Linux) mirall/2.7.0v270beta1 (build 1913) (cernboxcmd, centos-3.10.0-1127.18.2.el7.x86_64 ClientArchitecture:x86_64 OsArchitecture:x86_64),
      "Accept":"*/*",
      "X-Request-ID":7969c301-21f2-485f-95a2-31e5905d994a
    }
  Data: [8 bytes of application/offset+octet-stream data]"

09-07 09:54:08:049 [ info sync.httplogger ]:	"7969c301-21f2-485f-95a2-31e5905d994a: 
  Response: POST https://<user>:<pass>@<server>/remote.php/dav/files/<user>/ 
  Header: 
    {
      "Server":nginx/1.17.10,
      "Date":"Mon,07 Sep 2020 07:54:08 GMT",
      "Content-Type":"application/octet-stream",
      "Content-Length":0,
      "Connection":"keep-alive",
      "Access-Control-Allow-Headers":"Tus-Resumable",
      "Upload-Length",
      "Upload-Metadata",
      "If-Match",
      "Access-Control-Allow-Origin":"*",
      "Access-Control-Expose-Headers":"Tus-Resumable",
      "Location",
      "Content-Security-Policy":"default-src""none"";",
      "Etag":"\\""\"682724914095980544:0eb702f7\", 
      Last-Modified: Mon, 07 Sep 2020 07:54:07 +0000, 
      Location: https://<server>/data/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNTk5NTUxNjQ3LCJpYXQiOjE1OTk0NjUyNDcsInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE2MC9kYXRhL2YzYjU2OTY4LTUwNWYtNDRjYS1iNWQ2LTk0YjJkMDI2Zjc1NSJ9.w_nq7ixg6wdDsEaZc1P5A7NekgeAWGWUVy4A9Jdd4QU, Oc-Etag: \"682724914095980544:0eb702f7\", Oc-Fileid: MTI4NGQyMzgtYWE5Mi00MmNlLWJkYzQtMGIwMDAwMDA5MTU4OjY4MjcyNDkxNDA5NTk4MDU0NA==, Tus-Resumable: 1.0.0, Upload-Offset: 8, Vary: Origin, X-Access-Token: <access-token>, X-Content-Type-Options: nosniff, X-Download-Options: noopen, X-Frame-Options: SAMEORIGIN, X-Permitted-Cross-Domain-Policies: none, X-Robots-Tag: none, X-Xss-Protection: 1; mode=block, Strict-Transport-Security: max-age=15724800; includeSubDomains, 
    } 
  Data: []"
```

**Note:** this hasn't been tested with the regular (not-cli) client due to https://github.com/owncloud/product/issues/143

cc/ @TheOneRing

**Comment by TheOneRing** [comment]
@butonic is the client required to provide the dav endpoint?

**Comment by TheOneRing** [comment]
Fixed in #8082
